### PR TITLE
Add `mixins` module to `asyncio`

### PR DESCRIPTION
**Pull request by sobolevn**

### CODE REVIEW

**Comment by sobolevn** [comment]
I am checking what CI will say about adding module to `VERSIONS`, but not to the module. Source: https://github.com/python/cpython/commits/main/Lib/asyncio/mixins.py

**Comment by sobolevn** [comment]
This module does not have any public objects, but we can still add several protected ones. I don't know how they are supposed to be used though.

File: stdlib/VERSIONS [hunk 1]
```diff
@@ -62,6 +62,7 @@ array: 2.7-
 ast: 2.7-
 asynchat: 2.7-
 asyncio: 3.4-
+asyncio.mixins: 3.10-
```
Comment (sobolevn) [inline stdlib/VERSIONS:65]: `stubtest` is still failing for <3.9 🤔 
https://github.com/python/typeshed/runs/4686388253?check_suite_focus=true

**Comment by sobolevn** [comment]
@srittau @Akuli can you please help me? I don't understand why tests fail:

```

Run python tests/stubtest_stdlib.py
error: asyncio.mixins failed to find stubs
Stub:
MISSING
Runtime:
None
```

It happens on all python versions `<3.10`, when this module was introduced. But, I've marked it appropriately in `VERSION` file 🤔

File: stdlib/VERSIONS (hunk 1 above)
Comment (srittau) [reply stdlib/VERSIONS:65]: @sobolevn In theory, mypy should support this: python/mypy#11069, but in practice, `asyncio.exceptions` is also ignored in the stubtest allowlists. You need to do the same with `asyncio.mixins` for now.

**Review by AlexWaygood — COMMENTED** [review]
Could you update the classes in `asyncio.locks` and `asyncio.queues` so that they inherit from `_LoopBoundMixin`, as they do at runtime?

File: stdlib/VERSIONS (hunk 1 above)
Comment (srittau) [reply stdlib/VERSIONS:65]: python/mypy#12062

File: stdlib/asyncio/mixins.pyi [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import threading
+from typing import Any
+
+_global_lock: threading.Lock
+_marker: object
+
+class _LoopBoundMixin:
+    def __init__(self, *, loop: Any = ...) -> None: ...
```
Comment (srittau) [inline stdlib/asyncio/mixins.pyi:8]: Maybe we should just skip the `loop` argument, It can't really be used. Then we can also remove `_marker` above.

**Review by srittau — APPROVED** [review]
See my comment above, but I'm fine either way.

**Comment by sobolevn** [comment]
@AlexWaygood sure, in an hour 👍

**Comment by sobolevn** [comment]
I went for `loop: NoReturn`. I think that it can be the best solution:
1. It should not require any `stubtest` hacks
2. It indicates that `loop` is not allowed

And also, looks like the change propsed by @AlexWaygood will require some extra tweaks, so I am going to leave this for some next PRs (while thinking about the best way to implement it).

**Pull request merged**